UM-USElab/gradecraft-development
# 3240 Add individual assignments to google calendar

### Status
**READY FOR REVIEW**

### Description
When a student/instructor/observer looks at an individual assignment, they should be able to click an "Add to Google Calendar" button that works exactly as Events do. If there is no start time for the assignment (only a due at datetime), then set it to be 30 minutes prior to the due time.

### Related PRs
N/A

### Todos

### Deploy Notes
N/A

### Gem dependencies
N/A

### Migrations
NO

### Steps to Test or Reproduce
As a student/observer, go to the assignments menu and click any assignment. In the Description tab, the user should notice that the "Add to Google Calendar" button. Upon clicking the button, the assignment will be copied to the user's corresponding Google Calendar if the user has the User Authorization linking a Google Account to their GradeCraft account.

As an instructor, go to the assignments menu and click any assignment. In the top right corner of the screen the user should notice the "Add to Google Calendar" button. Upon clicking the button, the assignment will be copied to the user's corresponding Google Calendar if the user has the User Authorization linking a Google Account to their GradeCraft account.

### Impacted Areas in Application
* Assignments

======================
Closes #3240
